packages/react-ui-components/src/components/SchemaForm/ArrayFieldTemplate/ArrayFieldTemplate.tsx

```diff
@@ -132,18 +162,8 @@ const DefaultFixedArrayFieldTemplate = ({
 
 			<div className={styles['array-item-list']} key={`array-item-list-${idSchema.$id}`}>
 				{items && items.map(DefaultArrayItem)}
+				{addButton({ canAdd, disabled, readonly, uiSchema, onAddClick })}
```

Comment:
seems better to add the validation of the addButton component before the call for it. So, I suggest this:

```
{ canAdd && AddButton(...) }
```